staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -462,18 +456,19 @@ func (cfgCtlr *configController) digestConfigObjects(newPLs []*flowcontrol.Prior
 	return suggestedDelay, utilerrors.NewAggregate(errs)
 }
 
-// makeFlowSchemaConditionPatch takes in a condition and returns the patch status as a json.
-func makeFlowSchemaConditionPatch(condition flowcontrol.FlowSchemaCondition) ([]byte, error) {
-	o := struct {
-		Status flowcontrol.FlowSchemaStatus `json:"status"`
-	}{
-		Status: flowcontrol.FlowSchemaStatus{
-			Conditions: []flowcontrol.FlowSchemaCondition{
-				condition,
-			},
-		},
-	}
-	return json.Marshal(o)
+func toFlowSchemaApplyConfiguration(fsUpdate fsStatusUpdate) *flowcontrolapplyconfiguration.FlowSchemaApplyConfiguration {
+	fs := fsUpdate.flowSchema
+
+	// the condition field holds the new condition we want to update.
+	// TODO: this will break when we have multiple conditions for a flowschema
+	cond := fsUpdate.condition
+	condApplyConfiguration := flowcontrolapplyconfiguration.FlowSchemaCondition()
+	condApplyConfiguration.WithType(cond.Type).WithStatus(cond.Status).WithReason(cond.Reason).
+		WithLastTransitionTime(cond.LastTransitionTime).
+		WithMessage(cond.Message)
+
+	return flowcontrolapplyconfiguration.FlowSchema(fs.Name).
+		WithStatus(flowcontrolapplyconfiguration.FlowSchemaStatus().WithConditions(condApplyConfiguration))
```

Comment:
```suggestion
	return flowcontrolapplyconfiguration.FlowSchema(fsUpdate.flowSchema.Name).
		WithStatus(flowcontrolapplyconfiguration.FlowSchemaStatus().
		    WithConditions(flowcontrolapplyconfiguration.FlowSchemaCondition().
		        WithType(fsUpdate.condition.Type).
		        WithStatus(fsUpdate.condition.Status).
		        WithReason(fsUpdate.condition.Reason).
		        WithLastTransitionTime(fsUpdate.condition.LastTransitionTime).
		        WithMessage(fsUpdate.condition.Message)
		)
	)
```

Reply:
i separated the condition from the train wreck, let me know if this is fine.

Reply:
I don't see why you didn't go all the way to a single statement.
But it's really not a big deal.